src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1745,17 +1745,25 @@ def _gen_guid():
 def _random_password(length):
     import random
     import string
-    random_source = string.ascii_letters + string.digits + string.punctuation
-    password = random.choice(string.ascii_lowercase)
-    password += random.choice(string.ascii_uppercase)
-    password += random.choice(string.digits)
-    password += random.choice(string.punctuation)
+    safe_punctuation = '@%_-+=:,.(){}[]<>'
+    random_source = string.ascii_letters + string.digits + safe_punctuation
+    alphanumeric = string.ascii_letters + string.digits
 
-    for i in range(length - 4):    # pylint: disable=unused-variable
+    password = random.choice(alphanumeric)
```

Comment:
A better choice is to use [`secrets.choice`](https://docs.python.org/3/library/secrets.html#secrets.choice) which is the the "most secure source of randomness" instead of `random.choice`.

One problem is that `secrets` is introduced in Python 3.6. I am not sure whether using it will cause problem in Python 3.5 for which we have already dropped support.  + @fengzhou-msft